src/receive_imf.rs

```diff
@@ -1292,8 +1292,18 @@ async fn add_parts(
         && !mime_parser.parts.is_empty()
         && chat_id.get_ephemeral_timer(context).await? != ephemeral_timer
     {
+        let chat_contacts =
+            HashSet::<ContactId>::from_iter(chat::get_chat_contacts(context, chat_id).await?);
+        let is_from_in_chat =
+            !chat_contacts.contains(&ContactId::SELF) || chat_contacts.contains(&from_id);
```

Comment:
So, if SELF isn't in the chat then we allow ephemeral timer changes from anyone? This seems surprising to me, I would have expected this to just use `is_contact_in_chat()`.